File: xijinhui_window/XiJinHuiWindow/Models/StockItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using XiJinHuiWindow.Utils;

namespace XiJinHuiWindow.Models
    {
        public class StockItem
        {
            [JsonPropertyName("id")]
            public int? Id { get; set; }

            [JsonPropertyName("quantity"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? Quantity { get; set; }

            [JsonPropertyName("before_quantity"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? BeforeQuantity { get; set; }

            [JsonPropertyName("after_quantity"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? AfterQuantity { get; set; }

            [JsonPropertyName("price"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? Price { get; set; }

            [JsonPropertyName("note")]
            public string Note { get; set; }

            [JsonPropertyName("created_at")]
            public string CreatedAt { get; set; }

            [JsonPropertyName("updated_at")]
            public string UpdatedAt { get; set; }

            [JsonPropertyName("mfp_date")]
            public string MfpDate { get; set; }

            [JsonPropertyName("exp_date")]
            public string ExpDate { get; set; }

            [JsonPropertyName("item_id")]
            public int? ItemId { get; set; }

            [JsonPropertyName("item")]
            public Ingredient Item { get; set; }
        }
}

File: xijinhui_laravel/app/Models/StockItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class StockItem extends Model
{
    use HasFactory;

    protected $fillable = [
        'item_id',
        'quantity',
        'before_quantity',
        'after_quantity',
        'note',
        'input_date',
        'mfp_date',
        'exp_date',
        'stock_id',
    ];

    public function item()
    {
        return $this->belongsTo(Ingredient::class, 'item_id');
    }
}
